src/filters/time/time.test.ts

```diff
@@ -8,85 +8,199 @@ describe('Time', () => {
+                result: [
+                    { dateTime: mockTime(0, 0), text: 'set alarm clock for', matched: '0:00' },
+                ],
+            },
+            {
+                case: 'attend meeting at 11:00',
+                result: [
+                    { dateTime: mockTime(11, 0), text: 'attend meeting', matched: 'at 11:00' },
+                ],
+            },
+            {
+                case: 'afternoon tea at 4PM',
+                result: [
+                    { dateTime: mockTime(16, 0), text: 'afternoon tea', matched: 'at 4PM' },
+                ],
+            },
+            {
+                case: 'go to cinema at 2 pm',
+                result: [
+                    { dateTime: mockTime(14, 0), text: 'go to cinema', matched: 'at 2 pm' },
+                ],
+            },
+            {
+                case: 'clean the house at 4 p.M.',
+                result: [
+                    { dateTime: mockTime(16, 0), text: 'clean the house', matched: 'at 4 p.M.' },
+                ],
+            },
+            {
+                case: 'get shopping done by 4p.m.',
+                result: [
+                    { dateTime: mockTime(16, 0), text: 'get shopping done', matched: 'by 4p.m.' },
+                ],
+            },
+            {
+                case: 'go to work at quarter to 4',
+                result: [
+                    { dateTime: mockTime(3, 45), text: 'go to work', matched: 'at quarter to 4' },
+                ],
+            },
+            {
+                case: 'go to work at quarter past 4',
+                result: [
+                    { dateTime: mockTime(4, 15), text: 'go to work', matched: 'at quarter past 4' },
+                ],
+            },
+            {
+                case: 'go to work at quarter to 4pm',
+                result: [
+                    { dateTime: mockTime(15, 45), text: 'go to work', matched: 'at quarter to 4pm' },
+                ],
+            },
+            {
+                case: 'go to work at half past 4',
+                result: [
+                    { dateTime: mockTime(4, 30), text: 'go to work', matched: 'at half past 4' },
+                ],
+            },
+            {
+                case: 'go to work at 20 past 4',
+                result: [
+                    { dateTime: mockTime(4, 20), text: 'go to work', matched: 'at 20 past 4' },
+                ],
+            },
+            {
+                case: 'go to work at 20 minutes past 4',
+                result: [
+                    { dateTime: mockTime(4, 20), text: 'go to work', matched: 'at 20 minutes past 4' },
+                ],
+            },
+            {
+                case: 'go to work at 20 min past 4',
+                result: [
+                    { dateTime: mockTime(4, 20), text: 'go to work', matched: 'at 20 min past 4' },
+                ],
+            },
+            {
+                case: 'go to work at 20 min to 12am',
```

Comment:
Match against minutes with no space? e.g. `20mins`

Reply:
I don't cater for incorrect grammar ;)